google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -391,6 +396,12 @@ private Object writeReplace() {
           case JSON:
             builder.set(fieldName).to(Value.json((String) value));
             break;
+          case PROTO:
```

Comment:
`case ARRAY` is not handling `PROTO` and `ENUM` types. Please check if it is needed.